packages/orm/mongoose/src/decorators/preHook.ts

```
@@ -43,28 +43,37 @@
- * @param options
  * @returns {Function}
  * @decorator
  * @class
+ * @param method
+ * @param fn
+ * @param options
  */
-export function PreHook<T = any>(method: string, fn?: MongoosePreHookCB<T> | MongooseHookOptions, options?: MongooseHookOptions): Function {
+export function PreHook<T = any>(method: MongooseMethods, fn: MongoosePreHookCB<T>, options?: MongooseHookOptions): ClassDecorator;
+export function PreHook<T = any>(method: MongooseMethods, options?: MongooseHookOptions): StaticMethodDecorator;
+export function PreHook<T = any>(method: MongooseMethods, ...decoratorArgs: any[]): Function {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)